### ISSUE TITLE
Swap resource and day header in week view.

### ISSUE THREAD

**Issue by Patrick-Spielbichler** [jquense/react-big-calendar#2008]
Hello everyone,

I am currently working on a timeslotting system for various rooms (resources) and have come to the following problem. In the week view the columns are first grouped by the resources and afterwards by the days. I want to have the inverse grouping (days ---> resources). How can I achieve this?

I have also looked at issue [#727](https://github.com/jquense/react-big-calendar/issues/727) where the group order is the same as I want to achieve. However I have no plan how it works. I suppose an earlier version of react-big-calendar had other default properties. The question is: Which property can change this behavior.

Here is example code:

```
import React from 'react';
import BigCalendar from 'react-big-calendar';
import moment from 'moment';
import "react-big-calendar/lib/css/react-big-calendar.css"

let resources = [
     {
        resourceId: 1,
        resourceTitle: 'Dave',
    },
    {
        resourceId: 2,
        resourceTitle: 'Steve',
    },
    {
        resourceId: 3,
        resourceTitle: 'Barry',
    },
    {
        resourceId: 4,
        resourceTitle: 'Gary',
    },
]
let events = [
    {
        id: 0,
        title: 'Board meeting',
        start: new Date(2018, 0, 29, 9, 0, 0),
        end: new Date(2018, 0, 29, 13, 0, 0),
        resourceId: 1,
        color:"#000000"
    },
    {
        id: 1,
        title: 'MS training',
        start: new Date(2018, 0, 29, 14, 0, 0),
        end: new Date(2018, 0, 29, 16, 30, 0),
        resourceId: 2,
        color:"#EEEEEE"
    },
    {
        id: 2,
        title: 'Team lead meeting',
        start: new Date(2018, 0, 29, 8, 30, 0),
        end: new Date(2018, 0, 29, 12, 30, 0),
        resourceId: 3,
        color:"#0f61a1",
        allDay:true,
    },
    {
        id: 11,
        title: 'Birthday Party',
        start: new Date(2018, 0, 30, 7, 0, 0),
        end: new Date(2018, 0, 30, 10, 30, 0),
        resourceId: 4,
        color:"#4A10F6",
        allDay:true,
    },
];

class Calendar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }

    render() {
        return (
            <BigCalendar
                events={events}
                defaultDate={new Date(2018, 0, 29)}
                step={60}
                culture={"en-GB"}
                resources={resources}
                resourceIdAccessor="resourceId"
                resourceTitleAccessor="resourceTitle"
                localizer={BigCalendar.momentLocalizer(moment)}
            />
        );
    }
}

export default (Calendar);
```

Here is a screenshot of the output I get.
![image](https://user-images.githubusercontent.com/89066902/129681892-57414e4c-8158-48e6-8e43-09c36e34a57f.png)

And the output I want to get.
![image](https://user-images.githubusercontent.com/89066902/129682018-51a079d3-196e-4bed-9fd3-e25e6fd51161.png)

I would be very happy to find a solution for this.

Thank you very much!

**Comment by cutterbl** [comment]
There is no default property for getting this type of display. You would have to write a custom Week and Day view to achieve this, using an existing view as a blueprint.

**Comment by Surgeon83b** [comment]
Does this also stands for customizing week view as at screenshot (need to display days of week at another format)
![cale](https://github.com/jquense/react-big-calendar/assets/89166001/737135a1-ae33-4935-8017-3a245a13f31e)

**Comment by cutterbl** [comment]
@Surgeon83b Check out the [dayFormat](http://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats-day-format) bit of the [formats](http://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--formats) prop

**Comment by MehdiGR** [comment]
@fs-nathan did you find a solution

**Comment by MehdiGR** [comment]
@fs-nathan could plz suggest similar repo that has similar aspect

**Comment by cutterbl** [comment]
@fs-nathan Have you written an alternate view? I'd love to see it. Have you considered a PR? Even if it was just an example in Storybook.

**Comment by MehdiGR** [comment]
@fs-nathan I've tested not thing changed still same problem

**Comment by MehdiGR** [comment]
> > @fs-nathan I've tested not thing changed still same problem
> 
> How did you test it? Could you share an example repo?

I've modified the TimeGrid.js with your code

**Comment by MehdiGR** [comment]
> > > > @fs-nathan I've tested not thing changed still same problem
> > > 
> > > 
> > > How did you test it? Could you share an example repo?
> > 
> > 
> > I've modified the TimeGrid.js with your code
> 
> It works for me. Could you make sure your code gets built properly?

here is my calendar component `const DragAndDropCalendar = withDragAndDrop(Calendar);
<DragAndDropCalendar
       
          localizer={localizer}
          dayLayoutAlgorithm="no-overlap"
          events={events}
          defaultView={Views.WEEK}
          selectable={true} // Enables event creation
          onSelectSlot={onSelectSlot}
          onSelectEvent={handleUpdateEvent}
          resources={resourceMap}
          resourceIdAccessor={(resource: any) => resource.resourceId}
          resourceTitleAccessor={(resource: any) => resource.resourceTitle}
          min={new Date(0, 0, 0, 9, 0, 0)} // 9.00 AM
          max={new Date(0, 0, 0, 23, 0, 0)} // 5.00 PM
          onNavigate={(date) => setSelectedDate(date)}
          date={selectedDate}
          defaultDate={selectedDate}
          views={["day", "week", "month"]}
          // views={{ week: CustomWeek }}
          step={60}
          timeslots={1}
          style={{ height: 750 }}
        />`  with your TimeGrid update

**Comment by dquietguy** [comment]
> @cutterbl @MehdiGR I'm so sorry I missed your comments. I've been too busy recently to check Github notifications. I made an example PR here about what I changed for my project: https://github.com/jquense/react-big-calendar/pull/2609/files?diff=split&w=1
> 
> This will completely swap resource and day in header column Demo picture below: ![Screenshot 2024-06-13 at 11 06 13](https://private-user-images.githubusercontent.com/99985579/339197070-66bc5aca-679c-41e7-95c5-08226235bbbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NzE0MzYsIm5iZiI6MTcyMzc3MTEzNiwicGF0aCI6Ii85OTk4NTU3OS8zMzkxOTcwNzAtNjZiYzVhY2EtNjc5Yy00MWU3LTk1YzUtMDgyMjYyMzViYmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE2VDAxMTg1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExOWM5YjRkYjFhZDRjZDIyZWQ5NjcxMjA5ZDM2NGEyYWIyNTRmODhiYTY5ZjEzNjUxMGNkZDhlYjZjYWE2ODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zN2xZoa_dehrSc2ugM1jKWK7nDkexZ7-15Gqg6MNQBI)
> 
> I customized the library a lots and even added `/build` folder so the PR would be a mess. That's why I created new fork and include the only change in `TimeGrid`
> 
> Thanks RBC for all the goodness. It's a great library we could have

@fs-nathan I tried your forked RBC with custom TimeGrid.js code, it also built properly but I still don't get the same output as yours, am I missing something?
![image](https://github.com/user-attachments/assets/8c6f6453-122d-46b6-86ba-6fb0a0f866e6)

**Comment by cutterbl** [comment]
@dquietguy I saw your PR on this. Did you see my comment to the PR?

**Comment by dquietguy** [comment]
> @dquietguy I saw your PR on this. Did you see my comment to the PR?

@cutterbl not sure what PR are you talking about, I haven't notice any comments though.

**Comment by cutterbl** [comment]
@dquietguy Sorry. Guessing that was @MehdiGR 's PR. @MehdiGR , did you see my comments in your PR?

**Comment by MehdiGR** [comment]
@cutterbl Yes i've seen it thanks

**Comment by almeidapaulooliveira** [comment]
I'm implementing the excellent component on a SaaS, and this new view is **exactly** what I'm looking for. I was able to fully customize without touching the main code, would be nice to have this feature in.

I will try the fork and see how it goes.

![Screenshot 2024-09-12 at 09 41 07](https://github.com/user-attachments/assets/b37dcb7a-ed8e-4a44-83e7-4a07fcfa698e)

**Comment by anhdd-kuro** [comment]
@almeidapaulooliveira 
Thank for your contributions ! 
Is there any update on [PR](https://github.com/jquense/react-big-calendar/pull/2652) soon ?

**Comment by almeidapaulooliveira** [comment]
Hi there! A pull request is available:
https://github.com/jquense/react-big-calendar/pull/2652

I’ve been testing it in my SaaS, and it’s working well. Unfortunately, I haven’t had time to finish the props story as @cutterbl requested.

That’s the only thing missing. Feel free to fork my repo if you’d like to try it out.

**Comment by cutterbl** [comment]
This is now available in v1.16
